libretroshare/src/serialiser/rstypeserializer.h

```diff
@@ -439,176 +454,222 @@ struct RsTypeSerializer
 		}
 	}
 
-	/// std::vector<T>
+//============================================================================//
+//                       Sequence containers                                  //
+//============================================================================//
+
+	/** std::list is supported */ template <typename... Args>
+	struct is_sequence_container<std::list<Args...>>: std::true_type {};
+
+	/** std::set is supported */ template <typename... Args>
+	struct is_sequence_container<std::set<Args...>>: std::true_type {};
```

Comment:
When you deserialize using push_back, it's not the same than using std::set::insert. It only works if the items are already sorted, right? It's easy to make a rogue peer that will send unsorted items and create bugs at its friends, or even later on if , std::set is used in some GXS items.

Reply:
The elements are inserted with this `member.insert(member.end(), elem);` which on `std::vector`, `std::list` etc. is equivalent to `push_back(elem)` while on `std::set` which keep it's own sorting internally and check if the element is not already present it is equivalent to `instert(elem)`. So no problem with ~friends~ which send things out of order ;)